runtime/v2/shim/util_windows.go

```diff
@@ -57,35 +57,31 @@ func SocketAddress(ctx context.Context, id string) (string, error) {
 
 // AnonDialer returns a dialer for a npipe
 func AnonDialer(address string, timeout time.Duration) (net.Conn, error) {
-	var c net.Conn
-	var lastError error
-	timedOutError := errors.Errorf("timed out waiting for npipe %s", address)
 	start := time.Now()
 	for {
 		remaining := timeout - time.Since(start)
 		if remaining <= 0 {
-			lastError = timedOutError
-			break
+			return nil, errors.Errorf("timed out waiting for npipe %s", address)
```

Comment:
The error checking is better, but the logic still seems odd to me. The `remaining` check could be done before the sleep is called and error out before sleeping with the real error. The `remaining` could just be initialized to `timeout` before the `for` loop.

Reply:
@kevpar - We added DialPipeContext to winio a while back. Lets revendor to the latest release of go-winio `v0.4.13` and then we can use a pattern more like this:
```
ctx, cancel := context.WithTimeout(context.Background(), timeout)
defer cancel()

// If there is nobody serving the pipe we limit the timeout for this case to
// 5 seconds because any shim that would serve this endpoint should serve it
// within 5 seconds.
serveTimer := time.NewTimer(5*time.Second)
defer serveTimer.Done()
for {
	c, err := winio.DialPipeContext(ctx, address)
	if err != nil {
		if os.IsNotExist(err) {
			select {
			case <-serveTimer.C:
				return nil, errors.Wrap(os.ErrNotExist, "pipe not found before timeout")
			default:
				// Wait 10ms for the shim to serve and try again.
				time.Sleep(10 * time.Millisecond)
				continue
			}
		} else if err == context.DeadlineExceeded {
			return nil, errors.Wrapf(err, "timed out waiting for npipe %s", address)
		}
		return nil, err
	}
	return c, nil
}
```

Reply:
Doesn't even seem like we need to revendor, we already have `DialPipeContext`. I updated to the code Justin suggested.